File: linked-lists/Cpp_C/students_launch.c
int countStudents(int* students, int studentsSize, int* sandwiches, int sandwichesSize) {
    int n = studentsSize;
    int zero = 0;
    int one = 0;
    for (int i; i < n; i++) {
        if (students[i] == 0) {
            zero++;
        }
        else {
            one++;
        }
    }
    for (int i; i < n; i++) {
        if (sandwiches[i] == 0 && zero > 0) {
            zero--;
        }
        else if (sandwiches[i] == 1 && one > 0){
            one--;
        }
        else {
            return n-i;
        }
    }
    return zero+one;
}

File: linked-lists/Python/students_launch.py
from collections import Counter


class Solution:
    def countStudents(self, students: list[int], sandwiches: list[int]) -> int:
        res = len(students)
        cnt = Counter(students)
        for s in sandwiches:
            if cnt[s] > 0:
                res -= 1
                cnt[s] -= 1
            else:
                return res
        return res
